rm-datasource/src/main/java/io/seata/rm/datasource/exec/DeleteExecutor.java

```diff
@@ -61,10 +60,13 @@ protected TableRecords beforeImage() throws SQLException {
     }
 
     private String buildBeforeImageSQL(SQLDeleteRecognizer visitor, TableMeta tableMeta, ArrayList<List<Object>> paramAppenderList) {
-        String whereCondition = buildWhereCondition(visitor, paramAppenderList);
+        String orgSql = visitor.getOriginalSQL().replaceAll("(\\\r\\\n|\\\r|\\\n|\\\n\\\r)", " ");
         StringBuilder suffix = new StringBuilder(" FROM ").append(getFromTableInSQL());
-        if (StringUtils.isNotBlank(whereCondition)) {
-            suffix.append(" WHERE ").append(whereCondition);
+        this.getParamAppenderList(paramAppenderList);
+        String where = " where ";
```

Comment:
I think the here should not use "substring" take care of the sql parser.

Reply:
done